posh-svn.psm1

```diff
@@ -1,15 +1,87 @@
-if (Get-Command svn -errorAction SilentlyContinue)
-{
-  Push-Location $psScriptRoot
-  . ./SvnUtils.ps1
-  . ./SvnPrompt.ps1
-  . ./SvnTabExpansion.ps1
-  Pop-Location
-
-  Export-ModuleMember -Function @(
+param([switch] $NoVersionWarn)
+
+if (Get-Module posh-svn) { return }
+
+$psv = $PSVersionTable.PSVersion
+
+if ($psv.Major -lt 3 -and !$NoVersionWarn) {
+    Write-Warning ("posh-svn support for PowerShell 2.0 is deprecated; you have version $($psv).`n" +
+    "To download version 5.0, please visit https://www.microsoft.com/en-us/download/details.aspx?id=50395`n" +
+    "For more information and to discuss this, please visit **TODO PR**`n" +
+    "To suppress this warning, change your profile to include 'Import-Module posh-svn -Args `$true'.")
+}
+
+# Refer to svn application command via this, as we alias it.
+$svn = $null
+
+# & $PSScriptRoot\CheckVersion.ps1 > $null
+New-TimingInfo -Name CheckVersion -Command {
+    $Global:SvnMissing = $false
+
+    $svn = Get-Command 'svn' -CommandType Application -TotalCount 1 -ErrorAction SilentlyContinue
+    if (!$svn) {
+        Write-Warning "svn application command could not be found. Please create an alias or add it to your PATH."
+        $Global:SvnMissing = $true
+        return
+    }
+
+    # HACK determine a minimum required version, 1.6.0 is a guess
+    $requiredVersion = [Version]'1.6.0'
```

Comment:
Do you want to bump this up? 1.6 is ancient.

Reply:
I don't know what a minimum "reasonable" version is, I figured the syntax worked on 1.6 though.  We can bump higher if you'd rather.